src/percy-agent-client/dom.ts

```diff
@@ -225,6 +226,38 @@ class DOM {
     })
   }
 
+  /**
+   * Capture in-memory canvas elements & serialize them to images into the
+   * cloned DOM.
+   *
+   * Without this, applications that have canvas elements will be missing and
+   * appear broken. The Canvas DOM API allows you to covert them to images, which
+   * is what we're doing here to capture that in-memory state & serialize it
+   * into the DOM Percy captures.
+   *
+   * It's important to note the `.toDataURL` API requires WebGL canvas elements
+   * to use `preserveDrawingBuffer: true`. This is because `.toDataURL` captures
+   * from the drawing buffer, which is cleared after each render by default for
+   * performance.
+   *
+   */
+  private serializeCanvasElements(clonedDOM: HTMLDocument): void {
+    function canvasToImage($canvas: any) {
+      const $image = clonedDOM.createElement('img')
+      const canvasId = $canvas.getAttribute('data-percy-element-id')
+      const $clonedCanvas = clonedDOM.querySelector(`[data-percy-element-id=${canvasId}]`) as any
+
+      $image.setAttribute('style', 'max-width: 100%')
+      $image.classList.add('percy-canvas-image')
+
+      $clonedCanvas.style = 'display: none'
```

Comment:
Why do we use `style` here for the cloned canvas but `setAttribute` for the image?

Reply:
Honestly, how do you feel about deleting the node? This is just carelessly making sure it doesn't somehow appear in snapshots (we don't need it, the image is the source of truth now)

Reply:
That's true, this is the clone after all, no reason to keep it around